### PR TITLE
Encoding cleanup

### PR DESCRIPTION
**Pull request by slimbuck**
This PR:
- simplifies handling of texture encoding logic in the standard material
- splits out encode, decode and spherical helper chunks
- removes the hideous `texture2DRGBM` shader functions
- skybox can now render any of the supported encodings

### CODE REVIEW

**Review by willeastcott — APPROVED** [review]
Really nice refactor - cool!

File: src/graphics/program-lib/chunks/common/frag/envAtlas.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+export default /* glsl */`
+// the envAtlas is fixed at 512 pixels. every equirect is generated with 1 pixel boundary.
+const float atlasSize = 512.0;
```
Comment (mvaligursky) [inline src/graphics/program-lib/chunks/common/frag/envAtlas.js:3]: not related to this PR, but don't we handle larger atlases now?

Comment (slimbuck) [reply src/graphics/program-lib/chunks/common/frag/envAtlas.js:3]: We do, but actually still layout them out as if they were 512.

**Review by mvaligursky — APPROVED** [review]
looks great. any problem with changed options - I suspect the old options would not be commonly used?

**Pull request merged**